management_app/views/course.py

```diff
@@ -2,22 +2,140 @@
 from flask import redirect
 from flask import render_template
 from flask import url_for
+from flask import request
+from werkzeug.utils import secure_filename
+import pandas as pd
+from datetime import date
 
 from management_app.views.auth import login_required
 from management_app.db import get_db
+from management_app.views.utils import download_file, upload_file, remove_upload_file, get_upload_filepath
 
 courses = Blueprint('courses', __name__, url_prefix='/courses')
 
-@courses.route('/')
+@courses.route('/offerings', methods=['GET'])
 @login_required
-def index():
-    db = get_db()
-    # TODO: should filter by the year selected from the frontend
-    courses = db.execute(
-            'SELECT year, quarter, user_name, course_title_id, course_sec, enrollment'
-            ' FROM scheduled_teaching st JOIN courses ON st.course_id = courses.course_id JOIN users ON st.user_id = users.user_id'
-            ' WHERE year = 2022'#, (,)
-    ).fetchall()
+def offerings():
+    if request.method == 'GET':
+        # generate year options
+        todays_date = date.today()  # creating the date object of today's date
+        years = list(range(2019, todays_date.year+1))  # [2019, 2020, 2021, 2022]
```

Comment:
Not sure if we have the year range starting from 2019 and that would be hard-coded. Can we render the range from what data we have in the database dynamically?